imports/ui/PreviewCard.jsx

```diff
@@ -0,0 +1,173 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import './css/PreviewCard.css';
+import Gauge from 'react-summary-gauge-2';
+import { lastOfferedSems, lastSem, getGaugeValues } from './js/CourseCard.js';
+import Review from './Review.jsx';
+
+/*
+  Filtered Result Component.
+*/
+
+export default class PreviewCard extends Component {
+  constructor(props) {
+    super(props);
+    // set gauge values
+    this.state = {
+      id: this.props.course._id,
+      rating: this.props.course.classRating,
+      ratingColor: "E64458",
+      diff: this.props.course.classDifficulty,
+      diffColor: "E64458",
+      workload: this.props.course.classWorkload,
+      workloadColor: "E64458",
+      topReview: {}
+    };
+
+    this.updateColors = this.updateColors.bind(this);
+
+  }
+
+  componentDidMount() {
+    this.updateColors();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps != this.props) {
+      this.setState({
+        id: this.props.course._id,
+        rating: this.props.course.classRating,
+        diff: this.props.course.classDifficulty,
+        workload: this.props.course.classWorkload,
```

Comment:
Not sure why we need this indirection when these fields are always the same as the props. Why don't we just use the props?
Doing this is an anti-pattern in React: https://reactjs.org/docs/react-component.html#componentdidupdate